file_system_db_util.py

```diff
@@ -0,0 +1,62 @@
+import os
+import uuid
+import glob
+from datetime import datetime
+
+
+def generate_filename(original_name, uid=None):
+    """
+    Generate a filename by combining the UID, timestamp, and original name.
+
+    Args:
+        original_name (str): The original name of the file.
+        uid (str, optional): The UID to be used. If not provided, a new UID will be generated.
+
+    Returns:
+        tuple: A tuple containing the generated filename and the UID used.
+    """
+    if not uid:
+        uid = str(uuid.uuid4())
+    timestamp = datetime.now().strftime("%H-%M-%S")
+    filename = f"{uid}_{timestamp}_{original_name}"
+    return filename, uid
+
+
+def extract_data_from_file_name(filename):
+    """
+       Extracts the UID, timestamp, and original name from a filename.
+
+       Args:
+           filename (str): The filename from which to extract the UID, timestamp, and original name.
+
+       Returns:
+           tuple: A tuple containing the extracted UID, timestamp, and original name.
+       """
+    split_parts = filename.split("_")
+    uid = split_parts[0]
+    timestamp = split_parts[1]
```

Comment:
A non-trivial constant value found inside list - It's hard to remember what's special about this index. Name this index to preserve clarity :)